torch/optim/lr_scheduler.py

```diff
@@ -21,6 +22,13 @@
 
 SAVE_STATE_WARNING = "Please also save or load the state of the optimzer when saving or loading the scheduler."
 
+
+class StepsOnFlag(enum.Flag):
+    """A flag which indicates when a _LRScheduler should be stepped and supports standard bitwise logic."""
+    BATCH = enum.auto()  # steps at the end of a batch
+    EPOCH = enum.auto()  # steps at the end of an epoch
```

Comment:
Note to self: The users would do
```python
if scheduler.steps_on() == StepsOnFlag.BATCH:
    scheduler.step()
```
```python
if scheduler.steps_on() == StepsOnFlag.EPOCH:
    scheduler.step()
```

Reply:
A couple clarifications:
* the method is decorated with `@property`, so no need to include the `()`
* The enum is actually a `Flag`, so using the bitwise and operator would likely be better practice

```python
for _ in range(num_epochs):
    for batch in dl:
        train_batch(batch)
        if scheduler.steps_on & StepsOnFlag.BATCH:
            scheduler.step()

    validate()
    if scheduler.steps_on & StepsOnFlag.EPOCH:
        scheduler.step()
```